ls1intum/Artemis
# `Development`: Refactor Git service

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
The `GitService` currently throws an `InterruptedException` instead of a domain-specific one.

### Description
All places where the exception is caught it is done so together with the other possible exceptions from the `GitService`. Therefore, not throwing an `InterruptedException` and instead throwing a domain-specific one (`GitApiException` that is thrown already for different errors) makes the code cleaner without losing information. Other exception types for different errors might still be thrown as appropriate, especially by other services depending on the `GitService`.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

Info:
- No user-visible functionality changed.
- The server-side logic didn’t change either. Just an exception that shoudn’t be thrown during normal operation anyway (interrupt during `Thread.sleep`) no longer gets bubbled up through the whole system.

⇒ So just click around a bit in programming exercises to make sure the Git operations still work. I sadly can’t give more precise testing steps as the `GitService` clone operation is used by nearly all programming exercise tasks. If some of them still work, then they all should. 

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| `GitService` | 76% | 85% |
